spatialmath/box.go

```diff
@@ -352,3 +355,67 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// TODO: add function description.
+func (b *box) ToPointCloud(options map[string]interface{}) ([]r3.Vector, error) {
+	dims := [3]float64{b.halfSize[0] * 2, b.halfSize[1] * 2, b.halfSize[2] * 2}
+	var faces [][]float64
+	// explain what this does
+	var iter float64
+	if options["resolution"] != nil {
+		iter = options["resolution"].(float64)
+	} else {
+		iter = 0.15 // default value or spacing
+	}
+	// which faces are these
+	for i := 0.0; i <= dims[0]; i += iter {
+		for k := 0.0; k <= dims[2]; k += iter {
+			p1 := []float64{i, 0, k}
+			p2 := []float64{i, dims[1], k}
+			faces = append(faces, p1, p2)
+		}
+	}
+	// which faces are these
+	for j := 0.0; j <= dims[1]; j += iter {
+		for k := 0.0; k <= dims[2]; k += iter {
+			p1 := []float64{0, j, k}
+			p2 := []float64{dims[0], j, k}
+			faces = append(faces, p1, p2)
+		}
+	}
+	// which faces are these
+	for i := 0.0; i <= dims[0]; i += iter {
+		for j := 0.0; j <= dims[1]; j += iter {
+			p1 := []float64{i, j, 0}
+			p2 := []float64{i, j, dims[2]}
+			faces = append(faces, p1, p2)
+		}
+	}
+	// explain what this does
+	rotMat := b.Pose().Orientation().RotationMatrix().mat
```

Comment:
What I would suggest here is: rather than having this here, make a separate function called something like `transformPointsToPose`. Then, you can construct a list of points for each geometry with the center at (0,0,0), and call this function with that list of points plus the pose of the geometry.

That way all points are moved by the same code, which will make debugging in the future much easier.

Reply:
done